src/NServiceBus.Core/Features/FeatureActivator.cs

```diff
@@ -122,9 +122,41 @@ static List<FeatureInfo> Sort(IEnumerable<FeatureInfo> features)
                 node.Visit(output);
             }
 
+            // Step 4: DFS to check if we have an directed acyclic graph
+            foreach (var node in allNodes)
+            {
+                if (DirectedCycleExistsFrom(node, new Node [] { }))
+                {
+                    throw new ArgumentException("Cycle in dependency graph detected");
+                }
+            }
+
             return output;
         }
 
+        static bool DirectedCycleExistsFrom(Node node, IEnumerable<Node> visitedNodes)
+        {
+            if (node.previous.Any())
+            {
+                if (visitedNodes.Any(n => n == node))
```

Comment:
@WilliamBZA  since u always need to validate all of them in the happy path you may as well make visitedNodes a list, use an add below and avoid the allocation of IEnumerable